olp-cpp-sdk-core/src/client/ApiLookupClientImpl.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ApiLookupClientImpl.h"
+
+#include <olp/core/logging/Log.h>
+#include "client/api/PlatformApi.h"
+#include "client/api/ResourcesApi.h"
+#include "repository/ApiCacheRepository.h"
+
+namespace olp {
+namespace client {
+
+namespace {
+constexpr auto kLogTag = "ApiLookupClientImpl";
+
+std::string FindApi(const std::string& hrn, const Apis& apis,
+                    const std::string& service, const std::string& version) {
+  auto it = std::find_if(apis.begin(), apis.end(), [&](const Api& obj) -> bool {
+    return (obj.GetApi() == service && obj.GetVersion() == version);
+  });
+  if (it == apis.end()) {
+    OLP_SDK_LOG_WARNING_F(kLogTag,
+                          "LookupApi(%s/%s) service not found, hrn='%s'",
+                          service.c_str(), version.c_str(), hrn.c_str());
+
+    return {};
+  }
+
+  OLP_SDK_LOG_DEBUG_F(
+      kLogTag, "LookupApi(%s/%s) found, hrn='%s', service_url='%s'",
+      service.c_str(), version.c_str(), hrn.c_str(), it->GetBaseUrl().c_str());
```

Comment:
Why don't you move these logs to the caller method and avoid taking HRN in as parameter.